edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java

```diff
@@ -1178,6 +1179,59 @@ public List getViewableStudentsForUser(String gradebookUid, String userId, List
 		
 	}
 
+	/**
+	 * Get a list of permissions defined for the given user based on section and role or all sections if allowed. 
+	 * This method checks realms permissions for role/section and is independent of the 
+	 * gb_permissions_t permissions.
+	 *
+	 * note: If user has the grade privilege, they are given the GraderPermission.VIEW_COURSE_GRADE permission to match
+	 * GB classic functionality. This needs to be reviewed.
+	 *
+	 * @param userUuid
+	 * @param siteId
+	 * @param role user Role
+	 * @return list of {@link org.sakaiproject.service.gradebook.shared.PermissionDefinition PermissionDefinitions} or empty list if none
+	 */
+	public List<PermissionDefinition> getRealmsPermissionsForUser(String userUuid,String siteId, Role role){
+
+		List<PermissionDefinition> permissions = new ArrayList<PermissionDefinition>();
+
+		if( this.getGradebookService().isUserAllowedToGrade(siteId,userUuid)){
+			//FIXME:giving them view course grade (this needs to be reviewed!!), 
+			//it appears in GB classic, User can view course grades if they have the ability to grade in realms
```

Comment:
I think the issue with the FIXME is that GBNG has a view course grades permission that doesn't exist in realms. GB classic allowed a user to view course grades if they have grading permissions. I made a comment that it needs to be reviewed as I am not sure what the direction is for GBNG, why the view course grade permission was added. Without it, TA's cannot view course grades unless they are given the privilege through GBNG permissions. The proper solution may be to add the permission to realms templates.

Steve, many thanks for merging this to master!